File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/07 - Nested Conditional Statements Exercise/05. Fishing Boat .py
budget = int(input())
season = input()
people = int(input())

price = 0

if season == "Spring":
    price = 3000
elif season == "Summer" or season == "Autumn":
    price = 4200
elif season == "Winter":
    price = 2600

if people <= 6:
    price -= price * 0.1
elif 7 <= people <= 11:
    price -= price * 0.15
elif people >= 12:
    price -= price * 0.25

if season == "Spring" or season == "Summer" or season == "Winter":
    if people % 2 == 0:
        price -= price * 0.05

if budget >= price:
    print(f"Yes! You have {(budget - price):.2f} leva left.")
else:
    print(f"Not enough money! You need {(price - budget):.2f} leva.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/10. УПРАЖНЕНИЕ ПО-СЛОЖНИ ПРОВЕРКИ/05. Fishing Boat.js
function fishingBoat(...input) {
    let budget = Number(input.shift());
    let season = input.shift();
    let fishermen = Number(input.shift());

    let rent = {
        'Spring': 3000,
        'Summer': 4200,
        'Autumn': 4200,
        'Winter': 2600
    };

    let price = rent[season];

    if (fishermen <= 6) {
        price *= 0.9;
    } else if (fishermen > 6 && fishermen <= 11) {
        price *= 0.85;
    } else {
        price *= 0.75;
    }

    if (fishermen % 2 == 0 && season != "Autumn") {
        price *= 0.95;
    }

    if (budget >= price) {
        console.log(`Yes! You have ${(budget - price).toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${(price - budget).toFixed(2)} leva.`);
    }
}

fishingBoat(['3000', 'Summer', '11']);  // Expected Output:
// Not enough money! You need 570.00 leva.

fishingBoat(['3600', 'Autumn', '6']);  // Expected Output:
// Not enough money! You need 180.00 leva.

fishingBoat(['2000', 'Winter', '13']);  // Expected Output:
// Yes! You have 50.00 leva left.